modules/call-dispatch/src/lib.rs

```diff
@@ -46,24 +44,41 @@ use sp_std::{marker::PhantomData, prelude::*};
 /// Spec version type.
 pub type SpecVersion = u32;
 
-/// Origin of the call on the target chain.
+/// Origin of a Call when it is dispatched on this (a.k.a target) chain.
```

Comment:
The reason why I think this makes sense is because this origin is really only used when a call is actually dispatched on the target chain. The usage on the source side is through `verify_sending_message()`, which is only used to check if the correct origin will be used on the target chain

Reply:
True, but I think the user on the source chain needs that type though, cause it's part of the payload sent to `message_lane::send_message` function. I guess it makes sense to split the source and target types though and make them more explicit as suggested in the other comment.